service/src/main/kotlin/fi/espoo/evaka/payments/PaymentService.kt

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2017-2022 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.payments
+
+import fi.espoo.evaka.shared.FeatureConfig
+import fi.espoo.evaka.shared.PaymentId
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.BadRequest
+import fi.espoo.evaka.shared.domain.HelsinkiDateTime
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+
+@Component
+class PaymentService(
+    private val integrationClient: PaymentIntegrationClient,
+    private val featureConfig: FeatureConfig
+) {
+    fun sendPayments(
+        tx: Database.Transaction,
+        now: HelsinkiDateTime,
+        user: AuthenticatedUser.Employee,
+        paymentIds: List<PaymentId>,
+        paymentDate: LocalDate,
+        dueDate: LocalDate
+    ) {
+        val seriesStart = featureConfig.paymentNumberSeriesStart ?: throw Error("paymentNumberSeriesStart not configured")
+        val payments = tx.readPaymentsByIds(paymentIds)
+
+        val notDrafts = payments.filterNot { it.status == PaymentStatus.DRAFT }
+        if (notDrafts.isNotEmpty()) {
+            throw BadRequest("Some payments are not drafts")
+        }
+
+        val units = tx.getPaymentUnitsByIds(payments.map { it.unit.id }).associateBy { it.id }
```

Comment:
This is not a big deal, but why not query the necessary unit data in the same query with the other payment data?

Reply:
Good idea, fixed in #2582